clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1006,7 +1004,14 @@ public void assign(Collection<TopicPartition> partitions) {
         fetcher.sendFetches();
 
         long now = time.milliseconds();
-        client.poll(Math.min(coordinator.timeToNextPoll(now), timeout), now);
+        long pollTimeout = Math.min(coordinator.timeToNextPoll(now), timeout);
+
+        client.poll(pollTimeout, now, new PollCondition() {
+            @Override
+            public boolean pollNeeded() {
+                return !fetcher.hasCompletedFetches();
+            }
+        });
```

Comment:
Am I missing something or was previous code not doing a check and act?

Also, the method documentation states that it does heat-beating, is that still true?